subsys/bluetooth/host/gatt.c

```diff
@@ -1173,7 +1173,7 @@ static void gatt_discover_next(struct bt_conn *conn, u16_t last_handle,
 	}
 
 	/* Stop if over the range or the requests */
-	if (params->start_handle >= params->end_handle) {
+	if (params->start_handle > params->end_handle) {
```

Comment:
Are you sure this doesn't end up causing the end_handle to be discovered twice? Btw, bt_gatt_discover don't call this function directly, so you can give the same value to start_handle and end_handle.

Reply:
> Are you sure this doesn't end up causing the end_handle to be discovered twice?

The last discovered handle is incremented before this comparison (line 1172), we would only need an additional guard in case ```last_handle``` is equal to ```UINT16_MAX```. Something like this:

```
 	if (params->start_handle < UINT16_MAX) {
 		params->start_handle++;
 	} else {
 	 	goto done;
 	}
```

> Btw, bt_gatt_discover don't call this function directly, so you can give the same value to start_handle and end_handle.

I am aware of it, I described a potentially problematic situation here: https://github.com/zephyrproject-rtos/zephyr/issues/10105#issuecomment-423087695

Reply:
We found this issue while tracking down an issue where not all of the descriptors was not discovered in a central sample I am working on. This change fixed that, I have not yet discovered any problems that can be related to this change.

Reply:
I think the issue is with handle at UINT16_MAX, which is something PTS tests, but if we have the else that should still work.

Reply:
@Vudentz Sure. I will run PTS GATT Discovery related test cases tomorrow on this